scripts/dev-e2e.sh

```diff
@@ -153,31 +151,47 @@ main() {
     log "Armada is available"
   fi
 
-  echo "Creating $ARMADA_QUEUE queue..."
-  armadactl-retry create queue "$ARMADA_QUEUE"
+  armadactl-retry create queue "$ARMADA_QUEUE" 2>&1 | log_group "Creating $ARMADA_QUEUE queue"
 
-  echo "Loading Docker image $IMAGE_NAME into Armada cluster"
-  TMPDIR="$scripts/.tmp" "$AOHOME/bin/tooling/kind" load docker-image "$IMAGE_NAME" --name armada
+  (test -d "$scripts/.tmp" || mkdir "$scripts/.tmp";
+   TMPDIR="$scripts/.tmp" "$AOHOME/bin/tooling/kind" load docker-image "$IMAGE_NAME" --name armada 2>&1) \
+   | log_group "Loading Docker image $IMAGE_NAME into Armada cluster";
 
-  echo "Submitting SparkPI job"
-  PATH="$AOHOME/bin/tooling/:$PATH" "$scripts/submitSparkPi.sh"  2>&1 | tee submitSparkPi.log
+  # Pause to ensure that Armada cluster is fully ready to accept jobs; without this,
+  # proceeding immediately causes sporadic immediate job rejections by Armada
+  sleep 30
 
-  DRIVER_JOBID=$(grep '^Driver JobID:' submitSparkPi.log | awk '{print $3}')
-  EXECUTOR_JOBIDS=$(grep '^Executor JobID:' submitSparkPi.log | awk '{print $3}')
+  PATH="$scripts:$AOHOME/bin/tooling/:$PATH" JOBSET="$JOBSET" "$scripts/submitSparkPi.sh" 2>&1 | \
+    tee submitSparkPi.log
+  DRIVER_JOBID=$(grep '^Submitted driver job with ID:' submitSparkPi.log | awk '{print $6}' | sed -e 's/,$//')
+  EXECUTOR_JOBIDS=$(grep '^Submitted executor job with ID:' submitSparkPi.log | awk '{print $6}' | sed -e 's/,$//')
+
+  if [ "${GITHUB_ACTIONS:-false}" == "true" ]; then
+    echo "::group::submit"
+    echo "jobid=$DRIVER_JOBID" >> "$GITHUB_OUTPUT"
+    echo "::endgroup::"
+  fi
 
   if [ "$JOB_DETAILS" = 1 ]; then
-    sleep 5   # wait a moment for Armada to schedule & run the job
+    sleep 10   # wait a moment for Armada to schedule & run the job
 
-    timeout 1m "$scripts"/armadactl watch test driver --exit-if-inactive 2>&1 | tee armadactl.watch.log
-    if grep "Job failed:" armadactl.watch.log; then err "Job failed"; exit 1; fi
+    (timeout 3m "$scripts"/armadactl watch "$ARMADA_QUEUE" "$JOBSET" --exit-if-inactive 2>&1 | tee armadactl.watch.log; \
+     if grep "Job failed:" armadactl.watch.log; then err "Job failed"; exit 1; fi) | log_group "Watching Driver Job"
```

Comment:
We can print out the error outside the log group and thus simplify / improve readability of this expression:
```suggestion
    timeout 3m "$scripts"/armadactl watch "$ARMADA_QUEUE" "$JOBSET" --exit-if-inactive 2>&1 | tee armadactl.watch.log | log_group "Watching Driver Job"
    if grep "Job failed:" armadactl.watch.log; then err "Job failed"; exit 1; fi
```